### PR TITLE
Require manual Go version updates for patch versions

### LINKED ISSUE #5845: Go version upgrade script should not ignore patch version
This came up as part of #5837, specifically in the `docker/debug/Dockerfile` https://github.com/jaegertracing/jaeger/pull/5837/files#diff-ab8e812479730d5975aa87291a563e1dc0e0cf03c790d5d126c54f4248b3820bR4

When `scripts/check-go-version.sh` is run it only preforms substitution of major.minor versions, which resulted in `golang:1.22.6-alpine` becoming `golang:1.23.6-alpine` - a non-existent version at this time. In the other places it replaces versions like `1.22.x`, where substitution works fine, but when the patch version is explicitly specified the script should probably just fail and let us update the files manually.

### PR DESCRIPTION
**Pull request by wasup-yash**
## Which problem is this PR solving?
 Resolves #5845 

## Description of the changes
- check for the manual patch in docker file and notifies

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Comment by wasup-yash** [comment]
@yurishkuro tagging for the review as this is a draft and could move to review || close if not ok :)

File: scripts/check-go-version.sh [hunk 1]
```diff
@@ -79,7 +84,26 @@ function check() {
 check go.mod "^go\s\+$version_regex" "$go_previous_version"
 check internal/tools/go.mod "^go\s\+$version_regex" "$go_latest_version"
 
-check docker/debug/Dockerfile "^.*golang:$version_regex" "$go_latest_version"
+# Patch Version Handling in Dockerfile
+check_dockerfile() {
```
Comment (yurishkuro) [inline scripts/check-go-version.sh:88]: The original script was already detecting where changes were needed, it just applied them incorrectly. Why is this function needed?

Comment (wasup-yash) [reply scripts/check-go-version.sh:88]: yes, you are right, I thought to implement it specifically for docker as a function but yeah it is not needed. (also the incorrect is the sed command part? {too noob to detect the err})

**Comment by yurishkuro** [comment]
This is incorrectly failing now
```
./scripts/check-go-version.sh
go.mod                                             Go version: 1.22 
internal/tools/go.mod                              Go version: 1.23 
Patch version detected in docker/debug/Dockerfile (1.23.0). Manual update required.
```

**Comment by wasup-yash** [comment]
> This is incorrectly failing now
> 
> ```
> ./scripts/check-go-version.sh
> go.mod                                             Go version: 1.22 
> internal/tools/go.mod                              Go version: 1.23 
> Patch version detected in docker/debug/Dockerfile (1.23.0). Manual update required.
> ```

can you check now as I removed the docker func which was not needed.

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
Thanks!

**Comment by wasup-yash** [comment]
> Thanks!

he just want to ask a question which is not related to this PR as  [here](https://github.com/jaegertracing/jaeger/blob/main/CONTRIBUTING_GUIDELINES.md#:~:text=You%20do%20not%20need%20to%20squash%20the%20commits%2C%20it%20will%20happen%20once%20the%20PR%20is%20merged%20into%20the%20official%20repo%20(but%20each%20individual%20commit%20must%20be%20signed).) it is mentioned that the commits are automatically squashed so is there any script we are running for this in this repository wants to know it :)

**Pull request merged**